enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -2905,7 +2905,7 @@ func (c *FirecrackerContainer) VMConfig() *fcpb.VMConfiguration {
 }
 
 func (c *FirecrackerContainer) isBalloonEnabled() bool {
-	return *snaputil.EnableBalloon && c.recyclingEnabled
+	return *snaputil.EnableBalloon && c.recyclingEnabled && c.supportsRemoteSnapshots
 }
```

Comment:
could you include a comment on this function about why it's disabled for recycle and non-workflows?